object_store/src/throttle.rs

```diff
@@ -0,0 +1,500 @@
+
+        path
+    }
+
+    async fn place_test_objects(
+        store: &ThrottledStore<InMemory>,
+        n_entries: usize,
+    ) -> <ThrottledStore<InMemory> as ObjectStoreApi>::Path {
+        let mut prefix = store.new_path();
+        prefix.push_dir("foo");
+
+        // clean up store
+        for entry in store
+            .list(Some(&prefix))
+            .await
+            .unwrap()
+            .try_concat()
+            .await
+            .unwrap()
+        {
+            store.delete(&entry).await.unwrap();
+        }
+
+        // create new entries
+        for i in 0..n_entries {
+            let mut path = prefix.clone();
+            path.set_file_name(&i.to_string());
+
+            let data = Bytes::from("bar");
+            let stream_data = std::io::Result::Ok(data.clone());
+            let stream = futures::stream::once(async move { stream_data });
+            store.put(&path, stream, None).await.unwrap();
+        }
+
+        prefix
+    }
+
+    async fn measure_delete(store: &ThrottledStore<InMemory>, n_bytes: Option<usize>) -> Duration {
+        let path = place_test_object(store, n_bytes).await;
+
+        let t0 = Instant::now();
+        store.delete(&path).await.unwrap();
+        let t1 = Instant::now();
+
+        t1 - t0
+    }
+
+    async fn measure_get(store: &ThrottledStore<InMemory>, n_bytes: Option<usize>) -> Duration {
+        let path = place_test_object(store, n_bytes).await;
+
+        let t0 = Instant::now();
+        let res = store.get(&path).await;
+        if n_bytes.is_some() {
+            // need to consume bytes to provoke sleep times
+            res.unwrap()
+                .map_ok(|b| bytes::BytesMut::from(&b[..]))
+                .try_concat()
+                .await
+                .unwrap();
+        } else {
+            assert!(res.is_err());
+        }
+        let t1 = Instant::now();
+
+        t1 - t0
+    }
+
+    async fn measure_list(store: &ThrottledStore<InMemory>, n_entries: usize) -> Duration {
+        let prefix = place_test_objects(&store, n_entries).await;
+
+        let t0 = Instant::now();
+        store
+            .list(Some(&prefix))
+            .await
+            .unwrap()
+            .try_concat()
+            .await
+            .unwrap();
+        let t1 = Instant::now();
+
+        t1 - t0
+    }
+
+    async fn measure_list_with_delimiter(
+        store: &ThrottledStore<InMemory>,
+        n_entries: usize,
+    ) -> Duration {
+        let prefix = place_test_objects(&store, n_entries).await;
+
+        let t0 = Instant::now();
+        store.list_with_delimiter(&prefix).await.unwrap();
+        let t1 = Instant::now();
+
+        t1 - t0
+    }
+
+    async fn measure_put(store: &ThrottledStore<InMemory>, n_bytes: usize) -> Duration {
+        let mut path = store.new_path();
+        path.set_file_name("foo");
+
+        let data = Bytes::from_iter((0..n_bytes).map(|_| 1u8));
+        let stream_data = std::io::Result::Ok(data.clone());
```

Comment:
Same here as in `place_test_object`:

```suggestion
        let data = Bytes::from_iter(std::iter::repeat(1u8).take(n_bytes));
        let stream_data = std::io::Result::Ok(data);
```